rviz_plugin_tutorials/CMakeLists.txt

```
@@ -45,40 +40,49 @@ set(SRC_FILES
 ## library to be called ``${PROJECT_NAME}`` (which is
 ## "rviz_plugin_tutorials", or whatever your version of this project
 ## is called) and specify the list of source files we collected above
-## in ``${SRC_FILES}``.
-add_library(${PROJECT_NAME} ${SRC_FILES})
-
-## Link the myviz executable with whatever Qt libraries have been defined by
-## the ``find_package(Qt4 ...)`` line above, or by the
-## ``set(QT_LIBRARIES Qt5::Widgets)``, and with whatever libraries
-## catkin has included.
-##
-## Although this puts "rviz_plugin_tutorials" (or whatever you have
-## called the project) as the name of the library, cmake knows it is a
-## library and names the actual file something like
-## "librviz_plugin_tutorials.so", or whatever is appropriate for your
-## particular OS.
-target_link_libraries(${PROJECT_NAME} ${QT_LIBRARIES} ${catkin_LIBRARIES})
+## in ``${SRC_FILES}``. We also add the needed dependencies.
+add_library(${PROJECT_NAME} SHARED ${SRC_FILES})
+ament_target_dependencies(${PROJECT_NAME}
+  "geometry_msgs"
+  "rclcpp"
+  "rviz_common"
+  "rviz_rendering"
+  "sensor_msgs")
+target_include_directories(${PROJECT_NAME} PUBLIC
+  ${Qt5Widgets_INCLUDE_DIRS})
+
+## Here we export the plugins and meshes so they can be found by RViz at runtime.
+pluginlib_export_plugin_description_file(rviz_common plugin_description.xml)
+register_rviz_ogre_media_exports(DIRECTORIES "media")
+
 ## END_TUTORIAL
 
-## Install rules
+ament_export_dependencies(
+  geometry_msgs
+  rclcpp
+  rviz_common
+  rviz_rendering
+  sensor_msgs)
+
+if(BUILD_TESTING)
+  find_package(ament_lint_auto REQUIRED)
+  ament_lint_auto_find_test_dependencies()
+endif()
 
 install(TARGETS
   ${PROJECT_NAME}
-  ARCHIVE DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION}
-  LIBRARY DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION}
-  RUNTIME DESTINATION ${CATKIN_PACKAGE_BIN_DESTINATION}
+  ARCHIVE DESTINATION lib/${PROJECT_NAME}
+  LIBRARY DESTINATION lib/${PROJECT_NAME}
+  RUNTIME DESTINATION lib/${PROJECT_NAME}
 )
 
-install(FILES
-  plugin_description.xml
-  DESTINATION ${CATKIN_PACKAGE_SHARE_DESTINATION})
-
 install(DIRECTORY media/
-  DESTINATION ${CATKIN_PACKAGE_SHARE_DESTINATION}/media)
+  DESTINATION lib/${PROJECT_NAME}/media)
```

Maybe this should be `share`? I don't know if that affects runtime of the plugin.
```suggestion
  DESTINATION share/${PROJECT_NAME}/media)
```

---

Fixed